qemu/lib/cdrom.py

```diff
@@ -0,0 +1,102 @@
+"""
+cdrom event check utility functions.
+
+This module is meant to reduce code size by performing extra event check
+procedures. Generally, functions here should provide layers of extra event
+check for cdrom related operations.
+"""
+import logging
+from six import py2
+from functools import wraps
+from virttest import virt_vm, utils_misc
+
+def get_device_tray_status(vm, device_id):
+    """
+    check whether specified block device tray is open or not.
+    Return True, if device is opened, else False.
+
+    : param vm: VM object
+    : device_id: block device identifier
+    """
+    blocks_info = vm.monitor.info('block')
+
+    if device_id not in str(blocks_info):
+        raise ValueError('Device id: %s not recognizable by vm' % device_id)
+
+    if isinstance(blocks_info, str):
+        open_str = 'tray open'
+        close_str = 'tray closed'
+        for block in blocks_info.splitlines():
+            if device_id in block:
+                if open_str in block:
+                    return True
+                elif close_str in block:
+                    return False
+    else:
+        for block in blocks_info:
+            if device_id in str(block) and block.get('tray_open'):
+                return block['tray_open']
+    return False
+
+def cdrom_event_check(check=True):
+    def onDecorator(func):
+        @wraps(func)
+        def onCall(*args, **kargs):
+            code = func.func_code if py2 else func.__code__
+            operation = code.co_name
+            assert operation in ['eject_cdrom', 'change_media'], \
```

Comment:
I rarely see `assert` statement in most programs except in self-test scripts, such as `./test.py`.
I found you use it extensively in your patches, so I did a quick research.

>  **Python’s assert statement is a debugging aid, not a mechanism for handling run-time errors.** The goal of using assertions is to let developers find the likely root cause of a bug more quickly. An assertion error should never be raised unless there’s a bug in your program.

https://stackoverflow.com/questions/5142418/what-is-the-use-of-assert-in-python?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa